harmony_netcdf_to_zarr/mosaic_utilities.py

```diff
@@ -0,0 +1,195 @@
+""" A module including functions to take many input NetCDF-4 datasets and
+    create aggregated dimensions, allowing for a single Zarr output.
+
+"""
+from datetime import timedelta
+from typing import List, Optional, Union
+
+from dateutil.parser import parse as parse_datetime
+
+from netCDF4 import Dataset, Group, Variable
+import numpy as np
+
+
+NetCDF4Attribute = Union[bytes, str, np.integer, np.floating, np.ndarray]
+
+seconds_delta = timedelta(seconds=1)
+minutes_delta = timedelta(minutes=1)
+hours_delta = timedelta(hours=1)
+days_delta = timedelta(days=1)
+
+time_delta_to_unit_map = {seconds_delta: 'seconds',
+                          minutes_delta: 'minutes',
+                          hours_delta: 'hours',
+                          days_delta: 'days'}
+
+time_unit_to_delta_map = {'seconds': seconds_delta,
```

Comment:
I think this is a fairly comprehensive of acceptable time units, as defined by the [CF-Conventions](https://cfconventions.org/cf-conventions/cf-conventions.html#time-coordinate) and UDUNITS. There are a few other units, but they are things like months, which are much more tricky to use. This dictionary contains at least all the `UDUNITS` options for seconds, minutes, hours and days.

Reply:
But some of our products have "delta_time". Would that come into play here?

Reply:
Do you mean that the dimension name is `delta_time`? If so, I think we're okay, as the name of the dimension should not affect things. What's the classes in this module are looking at is the format of the `units` metadata attribute. If that attribute looks like `<time unit> since <epoch>`, then this code should handle it (providing the time unit is seconds, minutes, hours or days).

Reply:
Oh okay - you are correct. The delta_time variables that I have encountered have units of "time since....". I think we are good. Thnx.